src/Database/LSMTree/Internal/Unsliced.hs

```diff
@@ -74,8 +75,14 @@ fromUnslicedKey x = SerialisedKey (fromUnsliced (to x))
 instance Show (Unsliced SerialisedKey) where
   show x = show (fromUnslicedKey x)
 
-instance Eq (Unsliced SerialisedKey) where
-  x == y = fromUnslicedKey x == fromUnslicedKey y
-
-instance Ord (Unsliced SerialisedKey) where
-  x <= y = fromUnslicedKey x <= fromUnslicedKey y
+instance Eq (Unsliced a) where
+  Unsliced ba1 == Unsliced ba2 = SBS ba1' == SBS ba2'
+    where
+      !(ByteArray ba1') = ba1
+      !(ByteArray ba2') = ba2
+
+instance Ord (Unsliced a) where
+  compare (Unsliced ba1) (Unsliced ba2) = compare (SBS ba1') (SBS ba2')
+    where
+      !(ByteArray ba1') = ba1
+      !(ByteArray ba2') = ba2
```

Comment:
Nice, but I don't think we need the generalisation to any `a`. Any `a` might have its own `Ord`/`Eq` instance that deviates from `Ord`/`Eq` on short bytestrings. And we're currently only using it at `SerialisedKey` anyway